# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/devil/hotfix-libdir.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Install the libraries and pkg-config files into the configured libdir.

--- DevIL.orig/CMakeLists.txt	2017-01-02 01:03:56.000000000 +0100
+++ DevIL/CMakeLists.txt	2026-09-27 18:34:03.334257016 +0200
@@ -1,6 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 
 project(ImageLib)
+include(GNUInstallDirs)
 
 add_subdirectory(src-IL)
 add_subdirectory(src-ILU)
--- DevIL.orig/src-IL/CMakeLists.txt	2017-01-02 01:03:56.000000000 +0100
+++ DevIL/src-IL/CMakeLists.txt	2026-09-27 18:34:03.336233035 +0200
@@ -248,14 +248,14 @@ configure_file( pkgconfig/IL.pc.cmake.in
 
 # Installation
 install (TARGETS IL
-    ARCHIVE DESTINATION lib
-    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION bin
 )
 install (FILES ../include/IL/il.h DESTINATION include/IL)
 
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/IL.pc
-	DESTINATION lib/pkgconfig
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
 )
 
--- DevIL.orig/src-IL/pkgconfig/IL.pc.cmake.in	2017-01-02 01:03:56.000000000 +0100
+++ DevIL/src-IL/pkgconfig/IL.pc.cmake.in	2026-09-27 18:34:03.339209964 +0200
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
 includedir=${prefix}/include
-libdir=${exec_prefix}/lib
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 
 Name: IL
 Description: Developer's Image Library - IL part (just image load/save)
--- DevIL.orig/src-ILU/CMakeLists.txt	2017-01-02 01:03:56.000000000 +0100
+++ DevIL/src-ILU/CMakeLists.txt	2026-09-27 18:34:03.340530505 +0200
@@ -74,14 +74,14 @@ endif(UNICODE)
 
 # Installation
 install (TARGETS ILU
-    ARCHIVE DESTINATION lib
-    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION bin
 )
 install (FILES ../include/IL/ilu.h DESTINATION include/IL)
 
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/ILU.pc
-	DESTINATION lib/pkgconfig
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
 )
 
--- DevIL.orig/src-ILU/pkgconfig/ILU.pc.cmake.in	2017-01-02 01:03:56.000000000 +0100
+++ DevIL/src-ILU/pkgconfig/ILU.pc.cmake.in	2026-09-27 18:34:03.341862194 +0200
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
 includedir=${prefix}/include
-libdir=${exec_prefix}/lib
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 
 Name: ILU
 Description: Developer's Image Library - ILU part (image library utilities)
--- DevIL.orig/src-ILUT/CMakeLists.txt	2017-01-02 01:03:56.000000000 +0100
+++ DevIL/src-ILUT/CMakeLists.txt	2026-09-27 18:34:03.343303378 +0200
@@ -72,14 +72,14 @@ endif(UNICODE)
 
 # Installation
 install (TARGETS ILUT
-    ARCHIVE DESTINATION lib
-    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION bin
 )
 install (FILES ../include/IL/ilut.h DESTINATION include/IL)
 
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/ILUT.pc
-	DESTINATION lib/pkgconfig
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
 )
 
--- DevIL.orig/src-ILUT/pkgconfig/ILUT.pc.cmake.in	2017-01-02 01:03:56.000000000 +0100
+++ DevIL/src-ILUT/pkgconfig/ILUT.pc.cmake.in	2026-09-27 18:34:03.344530862 +0200
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
 includedir=${prefix}/include
-libdir=${exec_prefix}/lib
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 
 Name: ILUT
 Description: Developer's Image Library - ILUT part (connection to higher level libraries)
